core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -96,6 +101,32 @@
                 }
             });
 
+    private static final boolean USE_ROUTE_CACHE = getBoolean("useRouteCache", true);
```

Comment:
SGTM. How about:
- `-DrouteCache=disable` : Turn off the cache
- `-DrouteCache=...` : Use the cache with user-defined specifications
- No system property: Use the cache with the default specification

(Considering `-DrouteCache=""` as `turn off` might be confused with `default`, so I picked 'disable' keyword.)